src/ConfigServiceProvider.php

```diff
@@ -71,49 +71,32 @@ public function __construct(
         $prefix = self::DEFAULT_PREFIX,
         $separator = self::DEFAULT_SEPARATOR
     ) {
-        $this->config = [];
-
-        $config = ($config instanceof ApplicationConfig) ? $config : new ApplicationConfig($config, $separator);
+        $this->prefix = $prefix;
+        $this->config = ($config instanceof ApplicationConfig) ? $config : new ApplicationConfig($config, $separator);
```

Comment:
@tomphp 

Not sure about ternary statements of sorts, when they become too long I find legibility is increased when not using them, what do you think?

```php
if (!$config instanceof ApplicationConfig) {
    $config = new ApplicationConfig($config, $separator);
}

$this->prefix = $prefix;
$this->config = $config;
```

Also, it's easier to see coverage.

But, it's a matter of taste, so up to you, of course!